### ISSUE TITLE
How to use multiple encoding in multipart/form-data ?

### ISSUE THREAD

**Issue by mbachircherif** [apple/swift-openapi-generator#594]
### Question

Hi !

I have tried to upload an image that is in a different supported format (png or jpeg), but It didn't work for me.

Here's a demo of a requestBody that I used to achieve the following according to [OpenAPI encoding specification](https://spec.openapis.org/oas/v3.1.0#encoding-object):
```
requestBody:
description: The avatar file
  required: true
  content:
    multipart/form-data:
      schema:
        type: object
        properties:
          file:
            type: string
      encoding:
        file:
          contentType: image/png, image/jpeg
```

Unfortunatly, I'm getting this error:

`Validation warning: Inconsistency encountered when parsing ``: 'image/png, image/jpeg' could not be parsed as a Content Type. Content Types should have the format '<type>/<subtype>'. [context: codingPath=.paths['/profiles/{id}/avatar'].post.requestBody.content['multipart/form-data'].encoding.file.contentType, contextString=, subjectName=]warning: Validation warning: Inconsistency encountered when parsing ``: 'image/png, image/jpeg' could not be parsed as a Content Type. Content Types should have the format '<type>/<subtype>'. [context: codingPath=.paths['/profiles/{id}/avatar'].post.requestBody.content['multipart/form-data'].encoding.file.contentType, contextString=, subjectName=]Error: Invalid content type string: 'image/png, image/jpeg', must have 2 components separated by a slash.`

Can you tell me what I did wrong ?

Thanks !

**Comment by czechboy0** [comment]
It seems that we (through using OpenAPIKit) don't currently support a list of `contentType`, please try again by removing the second content type and just keep one.

cc @mattpolzin If I'm reading this right, this should be valid: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#encoding-object

**Comment by mattpolzin** [comment]
Yeah, a comma separated string is indeed valid there according to the spec. I just created a bug ticket to track this [here](https://github.com/mattpolzin/OpenAPIKit/issues/378).

**Comment by mbachircherif** [comment]
Hi,

Thanks for your prompt response.
Noted, I'll be following up till It's fixed ;)

**Comment by mattpolzin** [comment]
This has been fixed upstream as of [OpenAPIKit 3.2.0](https://github.com/mattpolzin/OpenAPIKit/releases/tag/3.2.0). Parsing should be fixed just by updating; actually using multiple content types will require the new `OpenAPIKit.Content.Encoding` property `contentTypes`.

The next alpha release of OpenAPIKit 4.0.0  (in the works right now) will also contain the fix.

**Comment by simonjbeaumont** [comment]
@mbachircherif would you confirm this resolved your issue by running `swift package update`?

Thanks @mattpolzin! 🙏

**Comment by mbachircherif** [comment]
@simonjbeaumont I confirm, no more issue after update.
Thanks all for your work !

**Comment by czechboy0** [comment]
The OpenAPIKit fix should unblock parsing of the OpenAPI doc, and will mostly make this work. But not fully, because I suspect this way we only send the first content type, instead of allowing the adopter to choose from the list.

If we can confirm that's the case, we can use this issue to track adding fuller support.

**Comment by czechboy0** [comment]
The support in OpenAPIKit landed, for anyone interested in contributing the enhancement, you should be unblocked.
